app/controllers/books_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  before_action :find_single_book, only: %i[show edit update destroy]
+
+  def index
+    @books = Book.all
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def show
+    @reviews = Review.where(book_id: params[:id])
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def new
+    @book = Book.new
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def create
+    @book = Book.new(book_params)
+    if @book.save
+      flash[:success] = 'Book create completed'
+      redirect_to @book
+    else
+      flash[:error] = 'Book create fail!'
+      render :index
+    end
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def edit
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def update
+    if @book.update(book_params)
+      flash[:success] = 'Book update completed'
+      redirect_to @book
+    else
+      flash[:error] = 'Updating fail!'
+      render :index
+    end
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def destroy
+    begin
+      @book.destroy
```

Comment:
`@book.destroy` returns true or false same as `save` and `update`, so it would be better to check if the record is destroyed successfully or not